source/write/replace.txt

```diff
@@ -0,0 +1,213 @@
+is ``"Primola Restaurant"``:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/write/replace.rb
+      :start-after: start-replace-one
+      :end-before: end-replace-one
+      :language: ruby
+      :dedent:
+
+   .. output::
+      :language: console
+      :visible: false
+
+      Replaced 1 document(s)
+
+.. important::
+
+   The value of the ``_id`` field is immutable. If your replacement
+   document specifies a value for the ``_id`` field, it must be the same
+   as the ``_id`` value of the existing document or the driver raises a
+   ``WriteError``.
+
+Customize the Replace Operation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can pass a ``Hash`` object as a parameter to the ``replace_one`` method to set options to
+configure the replace operation. If you don't specify any options, the driver performs the replace
+operation with default settings.
+
+The following table describes the options that you can use to
+configure the replace operation:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Option
+     - Description
+
+   * - ``upsert``
+     - | Specifies whether the replace operation performs an upsert operation if no 
+         documents match the query filter. For more information, see :manual:`upsert
+         behavior </reference/method/db.collection.replaceOne/#upsert>`
+         in the {+mdb-server+} manual.
+       | Defaults to ``false``.
+
+   * - ``bypass_document_validation``
+     - | Specifies whether the update operation bypasses document validation. This lets you 
+         update documents that don't meet the schema validation requirements, if any 
+         exist. For more information about schema validation, see :manual:`Schema
+         Validation </core/schema-validation/#schema-validation>` in the {+mdb-server+} manual.
+       | Defaults to ``false``.
+
+   * - ``collation``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the {+mdb-server+} manual.
+
+   * - ``session``
+     - | Specifies the session to use for the operation. To learn more about sessions, see
+         :manual:`Client Sessions and Causal Consistency Guarantees </core/read-isolation-consistency-recency/#std-label-sessions>`
+         in the {+mdb-server+} manual.
+
+   * - ``hint``
+     - | Sets the index to use when matching documents. 
+         For more information, see the :manual:`hint statement
+         </reference/method/db.collection.replaceOne/#std-label-replace-one-hint>`
+         in the {+mdb-server+} manual.
+
+   * - ``let``
+     - | Provides a map of parameter names and values to set top-level
+         variables for the operation. Values must be constant or closed
+         expressions that don't reference document fields.
+
+The following code performs the same replace operation as above, but sets the ``upsert``
```

Comment:
[Style guide](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#std-term-above) - use "preceding" to locate information on a page
```suggestion
The following code performs the same replace operation as the preceding example, but sets the ``upsert``
```